app/controllers/api/v0/tags_controller.rb

```diff
@@ -13,7 +11,11 @@ def index
         per_page = (params[:per_page] || 10).to_i
         num = [per_page, 1000].min
 
-        @tags = @tags.where(id: params[:tag_ids]) if params[:tag_ids].present?
+        if params[:tag_ids].present?
```

Comment:
Comment, potentially blocking: 

I don't think this is the right controller. In the past, we tried to not have our frontend call the external API. For some time we had a mix of both approaches, but we changed internal code to use non-API controllers with endpoints that (also) respond to JSON. I think this separation still makes sense.

Reply:
@citizen428 for context could you write more about your perspective of how "this separation still makes sense"; I want to work to understand that context and also how this was already something reviewed and no one else raised that contextual concern.

Reply:
While we technically make no stability guarantees for v0 of the API, adding something to a public API that's also available for others to consume is a much bigger commitment than modifying a controller that is purely for internal consumption.

Additionally, public APIs are often much more generic because we can't anticipate how they are used, whereas we know exactly what we want for internal endpoints.

> how this was already something reviewed and no one else raised that contextual concern

The reviewers weren't aware of the work that happened in this area in 2020? 

Since I'm about to leave I have no strong opinion either way, maybe the whole decision should be revisited.

Reply:
@citizen428 pointed me to https://github.com/forem/forem/issues/6255

Reply:
In general I do agree with @citizen428's overall approach and concern. Before moving this to the internal tags controller, I'm hoping to get one question answered:

I could see this added bit of functionality being useful for the V0 api. With that, is the expectation here that this behavior would be live in both the internal tags controller AND the V0 api namespace?

Reply:
Proceed with adding to the V0 api and the internal tags controller.  A bonus feature that seems useful.

Reply:
> I could see this added bit of functionality being useful for the V0 api. With that, is the expectation here that this behavior would be live in both the internal tags controller AND the V0 api namespace?

While I agree that it's a useful addition, anything in v0 will almost automatically become the baseline of v1, so be mindful of not introducing too many things that then you may need to keep on supporting. :-) I think I'm leaning towards not adding it to v0 now, and instead looking for related feature requests when defining v1 to see if people even want this. 

If you do add it to the v0 API, please update the docs accordingly. They live in https://github.com/forem/forem-docs, specifically `api_v0.yml`.

---

Random thoughts, feel free to skip: I think there's a much more substantial discussion to be had about the future of Forem's API in the future. Given that at least so far we have decided that internal pages do not consume the public REST API and that there are many potential use cases for an external API that we probably can't even envision yet, I'd consider going with GraphQL for the public API (maybe through something like Hasura or postgraphile, maybe more hand-rolled with graphql-ruby because that may be easier for self-hosting). I'm surprised by this myself, as I was not really into GraphQL for a long time. But I find that for "platform APIs" where you don't just want to consume data but maybe build whole apps it's actually more enjoyable.

Reply:
ah that make sense! thanks -- I've restored the V0 tags controller 😄